stan/math/prim/prob/ordered_logistic_lpmf.hpp

```diff
@@ -69,17 +69,21 @@ template <bool propto, typename T_y, typename T_loc, typename T_cut>
 return_type_t<T_loc, T_cut> ordered_logistic_lpmf(const T_y& y,
                                                   const T_loc& lambda,
                                                   const T_cut& c) {
-  static const char* function = "ordered_logistic";
-
   using T_partials_return = partials_return_t<T_loc, T_cut>;
-  using T_partials_vec = typename Eigen::Matrix<T_partials_return, -1, 1>;
-
-  scalar_seq_view<T_loc> lam_vec(lambda);
-  scalar_seq_view<T_y> y_vec(y);
-  vector_seq_view<T_cut> c_vec(c);
+  using T_partials_array = Eigen::Array<T_partials_return, -1, 1>;
+  using T_cuts_val = value_type_t<scalar_type_t<T_cut>>;
```

Comment:
`value_type_t` on a potentially `fvar` input is used here for example.

Reply:
Isn't something like `partials_return_t` the normal way of doing this?

Reply:
Hmm, might be. I am not that familiar with metaprograms having partials or return type in the name. Those names all seem confusing to me.